app/main.py

```diff
@@ -2,0 +1,21 @@
+from typing import Union
+
+from app.cafe import Cafe
+from app.errors import VaccineError, NotWearingMaskError
+
+
+def go_to_cafe(friends: list, cafe: Cafe) -> Union[str, None]:
+    masks_needed = 0
+
+    for friend in friends:
+        try:
+            cafe.visit_cafe(friend)
+        except VaccineError:
+            return "All friends should be vaccinated"
+        except NotWearingMaskError:
+            masks_needed += 1
```

Comment:
Code Efficiency: In this part of the code, you are catching exceptions and then returning a string based on the type of exception. This is not a very efficient way of handling errors. Instead, you could add a custom error message when you raise the exceptions in the `visit_cafe` method of the `Cafe` class. This way, you can just catch the exception and return its message. This will make your code cleaner and more maintainable.